packages/xstate-wallet/src/store/store.ts

```diff
@@ -358,20 +367,21 @@ export class Store {
   }
 
   public addState = (state: SignedState) =>
-    this.backend.transaction(
-      'readwrite',
-      [ObjectStores.channels, ObjectStores.nonces, ObjectStores.privateKeys],
-      async () => {
-        const channelId = calculateChannelId(state);
-        const memoryChannelStorage =
-          (await this.backend.getChannel(channelId)) || (await this.initializeChannel(state));
-        // TODO: This is kind of awkward
-        state.signatures.forEach(sig => memoryChannelStorage.addState(state, sig));
-        await this.backend.setChannel(channelId, memoryChannelStorage.data());
-        this._eventEmitter.emit('channelUpdated', memoryChannelStorage);
-        return memoryChannelStorage;
-      }
-    );
+    this.backend
+      .transaction(
+        'readwrite',
+        [ObjectStores.channels, ObjectStores.nonces, ObjectStores.privateKeys],
+        async () => {
+          const channelId = calculateChannelId(state);
+          const memoryChannelStorage =
+            (await this.backend.getChannel(channelId)) || (await this.initializeChannel(state));
+          // TODO: This is kind of awkward
+          state.signatures.forEach(sig => memoryChannelStorage.addState(state, sig));
+          await this.backend.setChannel(channelId, memoryChannelStorage.data());
+          return memoryChannelStorage;
+        }
+      )
+      .then(entry => this._eventEmitter.emit('channelUpdated', entry));
```

Comment:
Might be worthwhile leaving a comment explaining that we want this outside the transaction scope.